_source/_posts/2024-04-30-spring-api-fga.md

```diff
@@ -0,0 +1,901 @@
+---
+layout: blog_post
+title: "Adding Fine-Grained Authorization (FGA) to a Spring Boot API with OpenFGA"
+author: jimena-garbarino
+by: contractor
+communities: [security,java]
+description: "How to add Fine-Grained Authorization (FGA) to a Spring Boot API using the OpenFGA Spring Boot starter"
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+
+Fine-grained authorization (FGA) refers to the capability of granting individual users permission to perform particular actions on specific resources. Effective FGA systems enable the management of permissions for a large number of objects and users. These permissions can undergo frequent changes as the system dynamically adds objects and adjusts access permissions for its users.
+
+OpenFGA is an open-source authorization system designed by Okta for developers, and adopted by the Cloud Native Computing Foundation (CNCF). It offers scalability and flexibility for the implementation of RBAC and ABAC authorization models, moving authorization logic outside application code, making it simpler to evolve authorization policies as complexity grows. In this guide, you will learn how to secure a Spring Boot document API with Auth0 and integrate Fine-Grained Authorization (FGA) into the document operations with OpenFGA.
```

Comment:
While you can do RBAC and ABAC, it's a Relationship Based Access Control (ReBAC) system.

Reply:
Added the lline.